### PR TITLE
feat: align Babbel few-shot selection

### PR DESCRIPTION
**Pull request by rmens**
## Summary

- always select eight recent Babbel examples when enough candidates exist
- target a three accepted / five editor-adjusted mix
- limit candidates to the last three months and exclude drafts, weak examples, duplicates, and the current WordPress post
- diversify by output length, edit score, original length, and sentence count
- label accepted and edited examples explicitly in the AI conversation
- remove the configurable few-shot count and clean up its legacy setting

## Testing

- `composer test`
- `vendor/bin/phpcs`
- `composer phpstan`
- `npm run lint`
- `composer validate --strict`
- `composer audit`
- `npm audit --audit-level=high`
- `shellcheck tests/e2e/run.sh`
- isolated E2E suite: 7 Playwright tests and 18 WordPress/Babbel scenarios with 252 assertions

## Summary by CodeRabbit

* **New Features**
  * AI-generated content now uses up to eight curated examples from the past three months.
  * Examples combine accepted and editor-adjusted content, include provenance details, and exclude the current post.
  * Selection improves variety across text length, sentence count, and editing history.

* **Removed**
  * Removed the configurable “Few-shot Examples” setting.

* **Documentation**
  * Updated documentation and translations to reflect the new behavior.

* **Tests**
  * Added regression coverage for selection, scoring, filtering, and edge cases.

### CODE REVIEW

File: tests/e2e/suite.php [hunk 1]
```diff
@@ -184,19 +183,35 @@ private function test_published_story_creation(): void {
...
+				array(
+					'post_id'             => $post_id + 1000,
+					'input'               => $example_input,
+					'output'              => $example_output,
+					'edit_score'          => 25.0,
+					'original_word_count' => 6,
+					'output_word_count'   => 5,
+					'sentence_count'      => 1,
+					'provenance'          => 'edited',
 				),
```
[the remark this answers is not in the document]
Comment (rmens) [reply tests/e2e/suite.php:222]: Fixed in 7f6a234. The E2E fixture now includes a non-current accepted example as well as the edited example. Assertions inspect the structured OpenAI request: for each provenance they verify the exact canonical label and input in a user message and the corresponding output in the immediately following assistant message. Verified against the real WordPress AI Client provider path in Docker: 7 Playwright tests and 18 PHP E2E scenarios with 264 assertions passed.

File: includes/few-shot-cache.php [hunk 2]
```diff
@@ -85,211 +79,430 @@ function sync_few_shot_examples(): void {
...
-function build_few_shot_candidates( array $stories ): array {
-	$candidates = array();
+function build_few_shot_candidates( array $stories, int $created_after_timestamp ): array {
+	$candidates    = array();
+	$seen_post_ids = array();
+
+	// Warm the post cache in one query instead of one query per get_post() call below.
+	$post_ids = array_filter( array_map( static fn ( array $story ): int => (int) ( $story['metadata']['wordpress_id'] ?? 0 ), $stories ) );
+	if ( array() !== $post_ids ) {
+		_prime_post_caches( array_values( $post_ids ), false, false );
```
[the remark this answers is not in the document]
Comment (rmens) [reply includes/few-shot-cache.php:124]: Fixed in 7f6a234 at the actual Babbel response boundary. babbel_fetch_recent_stories now validates that data is an array and filters it to array entries before returning, so neither the typed cache-warming callback nor the candidate loop can receive a scalar or null item. A malformed collection returns WP_Error, which also preserves the existing cache instead of treating a broken response as an empty successful sync.

File: includes/few-shot-cache.php [hunk 3]
```diff
@@ -85,211 +79,430 @@ function sync_few_shot_examples(): void {
...
+
+	if ( count( $selected ) < $max_count ) {
+		$selected_ids = array_fill_keys( array_column( $selected, 'post_id' ), true );
+		$remaining    = array_values(
+			array_filter(
+				$candidates,
+				static fn ( array $candidate ): bool => ! isset( $selected_ids[ $candidate['post_id'] ] )
+			)
+		);
+		$selected     = array_merge( $selected, select_diverse_examples( $remaining, $max_count - count( $selected ) ) );
```
[the remark this answers is not in the document]
Comment (rmens) [reply includes/few-shot-cache.php:428]: Fixed in 7f6a234. select_example_mix now applies the accepted-or-edited-score-at-least-1.0 eligibility rule once, before cohort selection. The fallback therefore reuses the same eligible pool instead of duplicating and potentially drifting from the cohort rule. Added a regression case where the mix is short and a 0.5-score edited candidate must still stay excluded.

File: tests/php/few-shot-selection.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	);
+	$ensure(
+		0.0 < KnabbelWP\calculate_edit_score( 'Eerste tekst.', 'Volledig andere tekst.' ),
+		'Changed text must receive a positive edit score.'
+	);
+	$ensure( 6 === KnabbelWP\few_shot_count_words( 'Er is 3,8 miljoen euro beschikbaar.' ), 'Decimal numbers must count as one word.' );
+	$ensure(
+		2 === KnabbelWP\few_shot_count_sentences( 'Het evenement begint vandaag. 365 kinderen doen mee.' ),
+		'A sentence starting with a number must be detected.'
+	);
```
[the remark this answers is not in the document]
Comment (rmens) [reply tests/php/few-shot-selection.php:109]: The still-relevant parts are covered now: fewer-than-eight was added in 676e10c, and 7f6a234 adds the weak-edited fallback regression. I am not adding age, draft, minimum-length, or duplicate fixtures to select_example_mix: those are enforced upstream by the Babbel query and build_few_shot_candidates, while the selector intentionally accepts the normalized FewShotCandidate shape, which has no date or status fields. Putting those rules into this selector test would duplicate responsibilities and could pass while the real API/build path regressed. The Docker sync scenario exercises that real path; a dedicated builder test would require a WordPress test harness that this repository does not currently have.

**Comment by rmens** [comment]
@coderabbitai full review

File: includes/ai-handler.php [hunk 5]
```diff
@@ -87,28 +89,22 @@ function ai_parse_model_setting( mixed $value ): ?array {
...
-		}
+	$messages = array();
+	foreach ( get_few_shot_examples( $current_post_id ) as $example ) {
+		$example_label = 'accepted' === $example['provenance']
+			? 'VOORBEELD DAT DE REDACTIE DIRECT HEEFT GEACCEPTEERD'
+			: 'VOORBEELD DAT DE REDACTIE HEEFT AANGEPAST';
+		$messages[]    = new UserMessage( array( new MessagePart( sprintf( AI_EXAMPLE_PROMPT, $example_label, $example['input'] ) ) ) );
+		$messages[]    = new ModelMessage( array( new MessagePart( $example['output'] ) ) );
 	}
 	$messages[] = new UserMessage( array( new MessagePart( sprintf( AI_ARTICLE_PROMPT, $content ) ) ) );
```
[the remark this answers is not in the document]
Comment (rmens) [reply includes/ai-handler.php:109]: I investigated this against the actual WordPress 7.0 AI Client and the official OpenAI provider and am deliberately not adding a fixed character budget. WordPress using_max_tokens() limits output only; ModelMetadata exposes capabilities/options but no context-window size or tokenizer, and the OpenAI provider metadata likewise has no input limit. A character constant therefore cannot reliably protect a provider context window, while the suggested break would discard all later (possibly smaller) examples. It would also leave the final article and system instruction unbounded, so it does not establish a total prompt bound. The existing eight-example count is already the intentional hard bound. Without an observed failure or an explicit cost/size requirement, another arbitrary cutoff is guesswork and YAGNI; once the platform exposes model-aware input limits, that native signal is the right place to enforce this.

File: README.md [hunk 6]
```diff
@@ -7,7 +7,8 @@ Publishes selected WordPress posts as radio stories in the [Babbel API](https://
 - adds a native **Radionieuws** control to the Publish box in the Classic Editor
 - generates provider-independent speech text through the WordPress AI Client
 - creates, updates, deletes, and restores the matching Babbel story
-- learns from editor-corrected Babbel text through nightly few-shot synchronization
+- uses a nightly mix of eight Babbel examples from the last three months: three directly accepted and five editor-adjusted texts when available
```
[the remark this answers is not in the document]
Comment (rmens) [reply README.md]: Fixed in 7737040. The README now says that the nightly job refreshes the three-month candidate pool and that eight examples are selected per AI request. The following bullet retains the current-post exclusion explicitly. This now matches sync_few_shot_examples() and get_few_shot_examples().

**Comment by rmens** [comment]
@coderabbitai full review

**Comment by rmens** [comment]
@coderabbitai full review
